sample_metadata/parser/generic_metadata_parser.py

```diff
@@ -202,23 +207,38 @@ def get_sequence_meta(self, sample_id: str, row: GroupedRow) -> Dict[str, Any]:
                 gvcf_filenames.extend(row[self.gvcf_column].split(','))
 
         # strip in case collaborator put "file1, file2"
+        full_filenames: List[str] = []
         if read_filenames:
-            full_filenames = [self.file_path(f.strip()) for f in read_filenames]
-            reads, reads_type = self.parse_file(full_filenames)
-
-            collapsed_sequence_meta['reads'] = reads
+            full_filenames.extend(self.file_path(f.strip()) for f in read_filenames)
+        if gvcf_filenames:
+            full_filenames.extend(self.file_path(f.strip()) for f in gvcf_filenames)
+
+        file_types = await self.parse_files(sample_id, full_filenames)
+        reads = file_types.get('reads')
+        variants = file_types.get('variants')
```

Comment:
It's a bit hard to read code with nested dictionaries. Perhaps explicit type hints could help to read it faster? Still, not a big deal, feel free to ignore.

```suggestion
        file_types: Dict[str, Dict[str, List]] = await self.parse_files(sample_id, full_filenames)
        reads: Dict[str, List] = file_types.get('reads')
        variants: Dict[str, List] = file_types.get('variants')
```